src/circuit/modes/evaluate_mode.rs

```diff
@@ -125,13 +125,12 @@ impl<H: GateHasher, SRC: CiphertextSource> CircuitMode for EvaluateMode<H, SRC>
         let a = self.lookup_wire(gate.wire_a).unwrap();
         let b = self.lookup_wire(gate.wire_b).unwrap();
 
-        let gate_id = self.next_gate_index();
-
         // If C is unreachable, skip evaluation and do not advance gate index.
         if gate.wire_c == WireId::UNREACHABLE {
             return;
         }
 
+        let gate_id = self.next_gate_index();
```

Comment:
It's a behavioral change that shouldn't be there.
Here you increment the `gate_id` depending on the input of the circuit (`vk`) so you will have a different topology each time, since `gate_id` is part of it.